src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -102,17 +99,18 @@ class BlockingProcess(
 
   private val process = startProcess()
 
+  // This shutdown hook is required because the stdio actors will be blocked on a read. The only
+  // reliable means of unblocking them is to destroy the thing they're listening to.
+  context.actorOf(ShutdownHook.props(() => process.destroy()), "process-destroyer")
```

Comment:
Please provide a solution to this problem - I'm all out of ideas. 

> Is the order of shutdown of the children is guaranteed, so the process will never get destroyed -before- the reader and writer have been shut down?

No, it is not guaranteed - just highly unlikely.

> Right now there's also 2 different calls to Process.destroy() both when responding to Destroy message and when the shutdownhook runs its postStop call. Why multiple calls?

One is for the regular use-case of terminating a process - this will be done in a known thread-safe manner. The other use-case is to catch actor system shutdown and tear everything down no matter what - it actually doesn't matter whether other threads for the parent actor hierarchy barf or not - the actor system is on its way out.

> I don't see how it would be possible to write a new Process API that would be platform independent and also non-blocking as the JDK API is blocking.

We'd have to resort to native code.

Anyhow, all that I can say is that, at least for now, this works. It is the *only* thing I've found to work, and I've been playing with this for a while. Please recommend a solution though. I'm all ears. The JDK is screwed when it comes to unblocking streams.